### PR TITLE
rename jit option for dotnet 8

### PR DESCRIPTION
**Pull request by kasperk81**

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]

File: lib/compilers/dotnet.ts [hunk 1]
```diff
@@ -312,19 +314,12 @@ class DotNetCompiler extends BaseCompiler {
...
-            '--codegenopt',
-            'JitDisasm=*',
-            '--codegenopt',
-            'JitDiffableDasm=1',
-            '--parallelism',
-            '1',
+            '-o', 'CompilerExplorer.r2r.dll',
+            '--codegenopt', (this.sdkMajorVersion < 7 ? 'NgenDisasm=*' : 'JitDisasm=*'),
+            '--codegenopt', (this.sdkMajorVersion < 8 ? 'JitDiffableDasm=1' : 'JitDisasmDiffable=1'),
+            '--parallelism', '1',
```
Comment (kasperk81) [inline lib/compilers/dotnet.ts:322]:
```suggestion
```

**Comment by kasperk81** [comment]
@mattgodbolt ptal.

btw, i found that the new versions are still not showing up

image

was there anything i missed in properties update last week or is it just a matter of time before the updates are deployed?

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
@kasperk81 the change hasn't been merged. I have no strong opinion, and I'm not sure why it didn't get merged - sorry. Doing so now. Will appear at some point over the weekend or Monday evening US time.

**Comment by kasperk81** [comment]
@mattgodbolt, now 7.0 has disappeared

image

was expecting to see all five versions
https://github.com/compiler-explorer/compiler-explorer/blob/610658c0b7346d2113dfe34da72862465ca75e69/etc/config/csharp.amazon.properties#L7

**Comment by mattgodbolt** [comment]
Nothing has been deployed yet

**Comment by mattgodbolt** [comment]
```
$ sudo ce_install install dotnet
2023-03-12 00:20:15,438 lib.installation_context INFO     Making uncached requests
2023-03-12 00:20:15,439 lib.ce_install  INFO     Creating multiprocessing pool with 2 workers
Installing compilers/dotnet v6.0.11
2023-03-12 00:20:22,762 lib.ce_install  INFO     compilers/dotnet v6.0.11 is already installed, skipping
Installing compilers/dotnet/newer v6.0.14
2023-03-12 00:20:22,762 lib.ce_install  INFO     compilers/dotnet/newer v6.0.14 is already installed, skipping
Installing compilers/dotnet/newer v7.0.1
2023-03-12 00:20:22,763 lib.ce_install  INFO     compilers/dotnet/newer v7.0.1 is already installed, skipping
Installing compilers/dotnet/newer v7.0.3
2023-03-12 00:20:22,763 lib.ce_install  INFO     compilers/dotnet/newer v7.0.3 is already installed, skipping
0 packages installed OK, 4 skipped, and 0 failed installation
```
all the compilers are installed ok...will deploy a new build shortly and then we can take stock of what's what.

**Comment by mattgodbolt** [comment]
I just pushed latest and now I only see .NET (main) and .NET 6.0.110, so there's clearly something going on here...

**Comment by mattgodbolt** [comment]
```
warn: Unable to stat dotnet701csharp compiler binary:  ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v7.0.1/.dotnet/dotnet' {"code":"ENOENT","errno":-2,"path":"/opt/compiler-explorer/dotnet-v7.0.1/.dotnet/dotnet","stack":"Error: ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v7.0.1/.dotnet/dotnet'","syscall":"stat"}
warn: Unable to stat dotnet703csharp compiler binary:  ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v7.0.3/.dotnet/dotnet' {"code":"ENOENT","errno":-2,"path":"/opt/compiler-explorer/dotnet-v7.0.3/.dotnet/dotnet","stack":"Error: ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v7.0.3/.dotnet/dotnet'","syscall":"stat"}
warn: Unable to stat dotnet6014csharp compiler binary:  ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v6.0.14/.dotnet/dotnet' {"code":"ENOENT","errno":-2,"path":"/opt/compiler-explorer/dotnet-v6.0.14/.dotnet/dotnet","stack":"Error: ENOENT: no such file or directory, stat '/opt/compiler-explorer/dotnet-v6.0.14/.dotnet/dotnet'","syscall":"stat"}
```
from our logs: https://github.com/compiler-explorer/infra/actions/runs/4394852664/jobs/7696175119

**Comment by mattgodbolt** [comment]
Those directories are empty it seems (!)

**Comment by mattgodbolt** [comment]
I see the problem and will fix...

**Comment by mattgodbolt** [comment]
Fixed - and now I'm in the process of rolling this out! LMK if you have further issues and sorry for thr trouble

**Comment by kasperk81** [comment]
thanks!

> 2023-03-12 00:20:22,762 lib.ce_install  INFO     compilers/dotnet v6.0.11 is already installed, skipping

could you revoke this cached installation and use the fresh tarball build for 6.0.11? the script (misc-builder/build/build-dotnet.sh) included version which we show on first line, but it is not picking it up due to caching:

👎 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 👍 
image

**Comment by mattgodbolt** [comment]
Will do.

**Comment by mattgodbolt** [comment]
I'm doing the revocation "in place" so it will unfortunately affect the live systems until the reinstall is done. We're working on making this not the case...

**Comment by jeremy-rifkin** [comment]
This is live

**Pull request merged**